manifests/job/present.pp

```diff
@@ -5,7 +5,10 @@
 # Parameters:
 #
 #   config
-#     the content of the jenkins job config file (required)
+#     the content of the jenkins job config file (template based)
+#
+#   temp_config_path
```

Comment:
yes, that's the idea, we used to call this `$generated` in our module so it could be something like `$generated_config_file`. I used `$temp_config_path` in reference to `$tmp_config_path` you're using hardcoded